### ISSUE TITLE
MetadataContainer incorrect

### ISSUE THREAD

**Issue by JZomDev** [kekolab/javaplex#1]
![image](https://github.com/kekolab/javaplex/assets/14265490/476a6d9e-059d-4081-9317-2129b5ea1855)

"Caused by: java.io.IOException: Cannot determine the right class to deserialize object. Object has no `composite´ attribute"

I don't quite know if this is plex's update or something not accounted for, how can I correct this?

FWIW since there isn't a gradle package for this, I copied the code here into my project without any edits.

**Comment by kekolab** [comment]
Hi @JZomDev,
thanks for using the library. I will look into it (although i cannot guarantee to do it fast, alas... I code for hobby).

That said, regarding your last line, you can use jitpack to integrate any github code into your build.gradle file. Have a look here: https://github.com/jitpack/jitpack.io

**Comment by kekolab** [comment]
May you please copy & paste all the stacktrace? Thanks

**Comment by JZomDev** [comment]
I am now using the jitpack of the link and it's neat but it doesn't seem to match what you have available, it also thinks 2.2.0 is the version to use. 

https://jitpack.io/com/github/kekolab/javaplex/

Within 2.2.0 PlexHTTPClient is not available and "main-55b48dab58-1" does not load

When I can recreate what I am doing with the proper lib I will come back with a stack trace but currently this is my blocker.

**Comment by kekolab** [comment]
Alas I have always used this repo as a test. Do not use 2.2.0 as it is pure crap. Soon I will publish a clean 3.0 version to use with jitpack based on the current SNAPSHOT. As soon as it's out, I will update this issue.
Then, when you have a stack trace I will check the issue.

From the image it seems like you're deserializing a PlexMediaServer (so far I've seen the node with title = action just in PlexMediaServer), but, theoretically the DirectoryDeserializer has no business in deserializing a PlexMediaServer as I ignore the deserializing of directories in PlexMediaServer object.

Doubtlessly that code is not clean and a check if the node has the "composite" attribute should be done before using it.

**Comment by kekolab** [comment]
BTW, with jitpack you can also integrate a tag or a release or a particular commit. If you want to use the current master branch snapshot, for example, this is the way to go:

```
implementation 'com.github.kekolab:javaplex:main-SNAPSHOT'
```

**Comment by JZomDev** [comment]
Maybe it'll be easier to explain what I am trying to do and how I am doing it.
the how:

I set up the PlexMediaServer (local) with my ip + port, client and Plex Key. (I know the server works because getCountryCode() works
I am getting the MetadataContainer using that server created.

I am trying to get the content of the MetadataContainer and that is what is getting the error thrown.

the what:
I am trying to see how many users are actively streaming content & if they're direct play or transcoding.

In the API if I check the MediaContainer's Metadata's Media's Part's Stream -> Width & Decision determines direct play / transcode.

```java


		PlexMediaServer server = getServer();
		MetadataContainer metadata = getMetadataContainer(server);

		List<> test = metadata.getMetadata();
```	

It is erroring within the assignment of test, ideally i could follow the content of mediacontainer -> metadata -> media -> .. and get a similar count through this lib instead of doing the json parsing from the GET that I can do.

**Comment by JZomDev** [comment]
> BTW, with jitpack you can also integrate a tag or a release or a particular commit. If you want to use the current master branch snapshot, for example, this is the way to go:
> 
> ```
> implementation 'com.github.kekolab:javaplex:main-SNAPSHOT'
> ```

I did try this but it fails to find artifact or something - i can either download the 1.0.0 jar or include the .zip directly in my code, really makes no difference to this.

**Comment by kekolab** [comment]
Alas, this library does not support (yet) statistics, which is what you're trying to get.
If you can give me the endpoint where you can find the statistics (something like http://IP:32400/SOMETHING/SOMETHINGELSE) I can try to integrate them.

Meanwhile, I will fix the bug so that in case there's no specialization to deserialize a PlexDirectory, it will default to the base class PlexDirectory and not throw an exception.

In addition here on GitHub you can find a python library much more advanced than this one which, I think, supports statistics.

**Comment by JZomDev** [comment]
> In addition here on GitHub you can find a python library much more advanced than this one which, I think, supports statistics.

Unfortunately my bread and butter is Java, I will just parse the API my self as I was doing prior to finding this to maybe make it easier to read. 

The end point I am using for stats is /status/sessions, it either is an empty json (no media playing) or it has an array for the content that is being played. 

I like the project and my intent was to leverage the tools to have a plex discord bot to have information about how many other people are using it without asking the admin

**Comment by JZomDev** [comment]
PFA for example response for a MediaContainer containing a single stream, 
[MediaContainer.json](https://github.com/kekolab/javaplex/files/14453831/MediaContainer.json)

**Comment by kekolab** [comment]
fixed by 51a887b

@JZomDev I created a fix (in a separate branch) for this issue. I will integrate and then publish it.

I will open another "feature request" issue to keep track of the request to access sessions.

**Comment by kekolab** [comment]
@JZomDev just to let you know that I have released a 3.1.0 version with support to /status/sessions.
I have also fixed the issues with jitpack.
